platform/darwin/test/MGLLightTest.mm

```diff
@@ -0,0 +1,211 @@
+#import <XCTest/XCTest.h>
+#import <Mapbox/Mapbox.h>
+
+#import "MGLLight_Private.h"
+
+#import "../../darwin/src/NSDate+MGLAdditions.h"
+
+#import <mbgl/style/light.hpp>
+#import <mbgl/style/types.hpp>
+#include <mbgl/style/transition_options.hpp>
+
+@interface MGLLightTest : XCTestCase
+
+@end
+
+@implementation MGLLightTest
+
+- (void)testProperties {
+    
+    MGLTransition defaultTransition = MGLTransitionMake(0, 0);
+    MGLTransition transition = MGLTransitionMake(6, 3);
+    mbgl::style::TransitionOptions transitionOptions { { MGLDurationFromTimeInterval(6) }, { MGLDurationFromTimeInterval(3) } };
+    
+    // anchor
+    {
+        auto light = new mbgl::style::Light();
+        MGLLight *mglLight = [[MGLLight alloc] initWithMBGLLight:light];
+        XCTAssertEqual(mglLight.lightAnchor, MGLLightAnchorViewport);
+        XCTAssertEqual(mglLight.lightAnchorTransition.delay, defaultTransition.delay);
+        XCTAssertEqual(mglLight.lightAnchorTransition.duration, defaultTransition.duration);
+        
+        auto lightFromMGLlight = [mglLight mbglLight];
+        
+        XCTAssertEqual(light->getDefaultAnchor(), lightFromMGLlight.getAnchor().asConstant());
+        auto anchorTransition = lightFromMGLlight.getAnchorTransition();
+        XCTAssert(anchorTransition.delay && MGLTimeIntervalFromDuration(*anchorTransition.delay) == defaultTransition.delay);
+        XCTAssert(anchorTransition.duration && MGLTimeIntervalFromDuration(*anchorTransition.duration) == defaultTransition.duration);
+        
+        mglLight.lightAnchor = MGLLightAnchorMap;
+        mglLight.lightAnchorTransition = transition;
+        
+        XCTAssertEqual(mglLight.lightAnchor, MGLLightAnchorMap);
+        XCTAssertEqual(mglLight.lightAnchorTransition.delay, transition.delay);
+        XCTAssertEqual(mglLight.lightAnchorTransition.duration, transition.duration);
+        
+        mbgl::style::PropertyValue<mbgl::style::LightAnchorType> anchorProperty = {  mbgl::style::LightAnchorType::Map  };
+        light->setAnchor(anchorProperty);
+        light->setAnchorTransition(transitionOptions);
+        
+        lightFromMGLlight = [mglLight mbglLight];
+        
+        XCTAssertEqual(light->getAnchor().asConstant(), lightFromMGLlight.getAnchor().asConstant());
+        anchorTransition = lightFromMGLlight.getAnchorTransition();
+        XCTAssert(anchorTransition.delay && MGLTimeIntervalFromDuration(*anchorTransition.delay) == transition.delay);
+        XCTAssert(anchorTransition.duration && MGLTimeIntervalFromDuration(*anchorTransition.duration) == transition.duration);
+        
+    }
+    
+    // position
+    {
+        auto light = new mbgl::style::Light();
```

Comment:
It seems odd that we're using the new keyword here. Would it work to declare `mbgl::style::Light light;` instead?